lib/ansible/executor/module_common.py

```diff
@@ -344,7 +348,10 @@ def debug(command, zipped_mod, json_params):
             exitcode = invoke_module(module, zipped_mod, ANSIBALLZ_PARAMS)
     finally:
         try:
-            shutil.rmtree(temp_path)
+            if remove_tempdir:
+                shutil.rmtree(tempdir)
```

Comment:
This might not be a good idea as it opens us up to a lot of race conditions.  I'd be tempted to just leave tempdir and only remove temp_path.  This means that when we are forced to makedir tempdir we need to give it relatively relaxed permissions.  (I think that's okay because temp_path has very strict permissions).

Reply:
yes, it should be the local dir created and not the one by the controller, i'll 'rephrase'